File: q0392/q392.cpp
class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        int ps = 0;
        int pt = 0;
        while (ps < s.size() && pt < t.size())
        {
            if (s[ps] == t[pt])
            {
                ps++;
            }
            pt++;
        }
        return ps == s.size();
    }
};

File: q0392/q392.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int ps = 0;
        int pt = 0;
        while (ps < s.length() && pt < t.length()) {
            if (s.charAt(ps) == t.charAt(pt)) {
                ps++;
            }
            pt++;
        }
        return ps == s.length();
    }
}